src/sinol_make/helpers/compiler.py

```diff
@@ -0,0 +1,109 @@
+import sinol_make.util as util
+import sys, subprocess
+
+def check_if_installed(compiler):
+	"""
+	Check if a compiler is installed
+	"""
+
+	try:
+		subprocess.call([compiler, '--version'], stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL)
+	except FileNotFoundError:
+		return False
+
+	return True
+
+
+def get_c_compiler(args = None):
+	"""
+	Get the C compiler
+	args - ArgParser arguments. Used to get compiler flags if set
+	"""
+
+	if 'c_compiler' in args and args.c_compiler is not None:
+		if not check_if_installed(args.c_compiler):
+			print(util.bold(util.color_red('C compiler \'' + args.c_compiler + '\' not found. You can set the C compiler with --c_compiler flag')))
+			exit(1)
+		else:
+			return args.c_compiler
+		
+	if sys.platform == 'win32' or sys.platform == 'linux':
+		if not check_if_installed('gcc'):
+			print(util.bold(util.color_red('C compiler \'gcc\' not found. You can set the C compiler with --c_compiler flag')))
+			exit(1)
+		else:
+			return 'gcc'
+	elif sys.platform == 'darwin':
+		for i in range(9, 13):
+			compiler = 'gcc-' + str(i)
+			if check_if_installed(compiler):
+				return compiler
```

Comment:
What if it's neither `win32`, `linux` and `darwin`?

Reply:
here are possible values of sys.platform https://docs.python.org/3.9/library/sys.html#sys.platform

Reply:
I forgot about cygwin, i don't think that aix is used anywhere

Reply:
In this case, we can add an `else assert` with a message that other operating systems are not used widely or not supported/documented, with the link you sent here.

Reply:
OK, there are a bit more possible values https://stackoverflow.com/questions/446209/possible-values-from-sys-platform